MainRepository/Program.cs

```diff
@@ -1,7 +1,19 @@
-﻿internal class Program
-{
-    private static void Main(string[] args)
-    {
-        Console.WriteLine("Hello, World!");
-    }
+﻿using System.Text;
+// Reduce string
+var inputString = new List<string> { "a", "bb", "aabb", "aabcbbaa", "aabbcdddc", "aacbbcdbda" };
+Dictionary<string, string> dictString = new () { { "a", "a" }, { "bb", "" }, { "aabb", "" }, { "aabcbbaa", "bc" }, { "aabbcdddc", "cdc" }, { "aacbbcdbda", "ccdbda" } };
+var reduceStringList = new List<string> ();

```

Comment:
If you make the method's return type as String, instead of void. This list storage could be avoided.